src/lib/provable/crypto/foreign-curve.ts

```diff
@@ -421,10 +425,10 @@ class ForeignCurveNotNeeded extends ForeignCurve {
  * const Curve = createForeignCurve(Crypto.CurveParams.Secp256k1);
  * ```
  *
- * `createForeignCurve(params)` takes curve parameters {@link CurveParams} as input.
+ * `createForeignCurve(params)` takes curve parameters `CurveParams` as input.
```

Comment:
Removing the link to `CurveParams` here so that `createForeignCurve` can remain public, and we don't also have to make `CurveParams` public.  This is my opinionated way of resolving, since the type `CurveParams` is linked in the function signature, it is quite a similar experience for the developer.